fud2/src/lib.rs

```diff
@@ -822,4 +822,60 @@ pub fn build_driver(bld: &mut DriverBuilder) {
             Ok(())
         },
     );
+
+    let cocotb_setup = bld.setup("cocotb", |e| {
+        e.config_var_or("cocotb-makefile-dir", "cocotb.makefile-dir", "$calyx-base/yxi/axi-calyx/cocotb")?;
+        // TODO (nate): this is duplicated from the sim_setup above. Can this be shared?
+        // The input data file. `sim.data` is required.
+        let data_name = e.config_val("sim.data")?;
+        let data_path = e.external_path(data_name.as_ref());
+        e.var("sim_data", data_path.as_str())?;
+
+        // Cocotb is wants files relative to the location of the makefile.
+        // This is annoying to calculate on the fly, so we just copy necessary files to the build directory
+        e.rule("copy", "cp $in $out")?;
+        e.rule("make", "make DATA_PATH=$sim_data VERILOG_SOURCE=$in COCOTB_LOG_LEVEL=CRITICAL > $out")?;
```

Comment:
Maybe this rule should be called `make-cocotb` or something (instead of the more generic `make`)?